### DISCUSSION TITLE
Customizable Layout

### DISCUSSION THREAD

**Discussion by r0b78** [directus/directus#2689] [Feature Requests]
Use bootstrap system of 12 slots and have mobile options also included 
![image](https://user-images.githubusercontent.com/22716876/80555693-44272980-898e-11ea-9c0e-0b2c77b09f21.png) In the short run maybe have another field option that use this red space, this area have huge potential on larger screens. Sometimes you have a text field, date field and a dropdown that you would like to have in the same row.

**Comment by benhaynes** [comment]
The 12 Column Grid idea is a good one, but since each interface needs to have _some_ minimum width, I worry about how much the form would change on different sized screens with fields oddly wrapping down at breakpoints.

However, I've been thinking about adding an option that allows the current 2-column form to become full-width... which would let you utilize that whitespace to the right. Another thing to consider is that in v9 (coming in a few weeks) we're using the right info sidebar a lot more... and are loading the page with it open when there's enough room. So there might not be as much whitespace in v9 to begin with.

Right now our App min width is already 1,264px (see component width calculation below)... and with a flex form that would still look good through 1,600px.

`64 + 220 + 32 + 300 + 32 + 300 + 32 + 284 = 1,264px viewport min width`

Lastly, we're adding some smart logic to interfaces that will allow them to go 3-up... which (when paired with the flex form) will kinda be what you're describing.

**Reply by r0b78** [reply]
That's a Thing When you have collections with a lot of fields you appreciate the space in the screen another space keeper is the sidebar, in desktop should be a place to hide it to make room to edit items in the collection so you could get all the screen for this. My thoughts for now are some option like this 
![image](https://user-images.githubusercontent.com/22716876/80600454-a82b0b80-89e9-11ea-9dd8-457777322ed1.png) When the component don't fit in the screen they go to full width like this 
![image](https://user-images.githubusercontent.com/22716876/80600538-c7299d80-89e9-11ea-9804-f852bc8b9889.png)

**Reply by benhaynes** [reply]
Yeah, the hard part of that is that difference between 1/3 and full. In your example, that last field (1/3) might be a smaller zip-code field (for example)... but then wraps down and becomes full-width. Now you have an address field that is smaller than the zip code... which feels weird.

Obviously Admins could learn this and work around it, but we're trying to use our UX experience to make the form setup a bit more fool-proof.

**Reply by rijkvanzanten** [reply]
I doubt there is a way to accurately and cleanly automatically wrap fields like that 🤔

**Reply by r0b78** [reply]
And some customization for use the bootstrap grid layout in the forms or is way to hard ?. This solve the problem of display on desktop, tablet and mobile way better by Admins.

**Reply by rijkvanzanten** [reply]
I'm not saying it's hard to implement on our end, I'm saying that the end result is inconsistent and will end up in a grid that's difficult to set up from settings, and that may or may not end up working properly (see ben's example above)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
